addons/purchase_stock/models/res_partner.py

```diff
@@ -52,9 +52,9 @@ def _compute_on_time_rate(self):
             ('purchase_line_id', 'in', order_lines.ids),
             ('state', '=', 'done')])
         # Fetch fields from db and put them in cache.
-        order_lines.read(['date_planned', 'partner_id', 'product_uom_qty'], load='')
+        order_lines.read(['date_promised', 'partner_id', 'product_uom_qty'], load='')
```

Comment:
I believe you need to do something similar in `VendorDelayReport`

Reply:
Changed for `VendorDelayReport` and Purchase (app) dashboard (within purchase list/kanban view).

It was decided that the Dashboard (app) dashboard would be left unchanged.